dds2val/README.md

```diff
@@ -0,0 +1,74 @@
+# DDS Provider
+
+The DDS provider provides data from an DDS middleware/API. For further understanding of the DDS middleware/API see [this](https://www.dds-foundation.org/what-is-dds-3/). The DDS provider only works with the KUKSA databroker. The KUKSA C++ server is not supported.
+
+## How to build
+
+### local build
+
+1. `python3 -m venv env && source env/bin/activate`
+2. `pip install -r requirements.txt`
+3. `chmod u+x ddsproviderlib/idls/generate_py_dataclass.sh`
+4. `./ddsproviderlib/idls/generate_py_dataclass.sh`
+
+### build image (suggested)
+
+1. `docker build -f Dockerfile --progress=plain --build-arg TARGETPLATFORM=linux/amd64 -t ddsprovider:latest .`
+
+### KML replay
+
+1. `python3 -m venv env && source env/bin/activate`
+2. `pip install -r requirements-kml.txt`
+3. `cd kml && python3 dds_kmlreplay.py directions.kml`
+
+## How to run
+
+Choose from local build or contanerization via docker.
+These steps are necessary:
+
+1. Run an instance of databroker aka: `docker run -it --rm --net=host ghcr.io/eclipse/kuksa.val/databroker:master`
+2. Start the KML replay with an active local python virtual environment: `pip install requirements/requirements-kml.txt && cd kml && python3 dds_kmlreplay.py directions.kml`
+3. Start the DDS provider with either: `docker run --rm -it --net=host ddsprovider:latest` or with an active local python virtual environment: `python3 ddsprovider.py`
+
+## Configure the DDS provider
+
+Configuration for the DDS provider is solved through setting environment variables. Have a look at the table below.
+
+| Environment variable          | default value | description                                                                                                                                      |
+| ----------------------------- | ------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ |
+| VEHICLEDATABROKER_DAPR_APP_ID | None          | DAPR ID for Vehicle App to look for. For more information to Vehicle Apps visit [Velocitas](https://eclipse-velocitas.github.io/velocitas-docs/) |
+| VDB_ADDRESS                   | 127.0.0.1     | Address where to look for (Vehicle) Databroker                                                                                                   |
+| DAPR_GRPC_PORT                | 55555         | Port where to look for (Vehicle) Databroker. If [DAPR](https://dapr.io/) gets used port of DAPR Sidecar where to look                            |
```

Comment:
We use this even if we do not use DAPR. We could possibly call it something else but that could possibly be part of another PR.

Reply:
I think we can handle it like in dbc feeder use DAPR_GRPC_PORT to overwrite port for vehicle databroker. So I will introduce another environment variable VDB_PORT